src/devices/src/virtio/block/device.rs

```diff
@@ -814,7 +815,53 @@ pub(crate) mod tests {
             check_metric_after_block!(
                 &METRICS.block.invalid_reqs_count,
                 1,
-                simulate_queue_event(&mut block, Some(true))
+                simulate_queue_and_async_completion_events(&mut block, true)
+            );
+
+            let used_idx = vq.used.idx.get();
+            assert_eq!(used_idx, 1);
+
+            let status_addr = GuestAddress(vq.dtable[2].addr.get());
+            assert_eq!(
+                mem.read_obj::<u8>(status_addr).unwrap(),
+                VIRTIO_BLK_S_IOERR as u8
+            );
+        }
+
+        // Read at out of bounds address.
```

Comment:
Can we also test for a Write operation at out of bounds address ?

Reply:
Done